src/clients/feed/feed.ts

```diff
@@ -161,6 +164,59 @@ class Feed {
     return this.makeStatusUpdate(itemOrItems, "seen");
   }
 
+  async markAllAsSeen() {
+    // To mark all of the messages as seen we:
+    // 1. Optimistically update *everything* we have in the store
+    // 2. We decrement the `unseen_count` to zero optimistically
+    // 3. We issue the API call to the endpoint
+    //
+    // Note: there is the potential for a race condition here because the bulk
+    // update is an async method, so if a new message comes in during this window before
+    // the update has been processed we'll effectively reset the `unseen_count` to be what it was.
```

Comment:
> I had thought that we could flag the store as not available to be written to or something and then poll to see when the job is completed, but that feels like an over-optimization for the time being and as such, I'm opting to ship this as is.

Maybe there's a dumber approach? For example, if we encounter this race condition (new message during bulk update window), would the following be feasible:

1. N new messages come in
2. We optimistically increment `un{seen,read}_count` from 0 to N
3. Done. No polling the batch update to resync the count state.

Reply:
Unfortunately here it's a bit more complex because when the real-time update comes in currently fetch the page of items. The race condition here isn't just about the badge counts, but about the items themselves. E.g. you could have archived everything, and then a new message is produced which causes a fetch before the archive all job is executed, and then you'd receive a list of feed items that includes things you previously thought were archived. This is a big edge case and something I think we can punt on for now.